client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,164 @@
+<script lang="ts">
+  import { authStore as user } from "$stores/auth";
+  import type { PrivatePlayer, Lobby } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+  import { getCurrentPlayer, getLobbyData, getPrivatePlayer, lobbyReturn } from "$lib/firebase/winloss";
+
+  export let lobbyCode: string;
+  let lobbyData: Lobby | undefined;
+  let currentRole: PrivatePlayer | undefined;
+  let lobbyRoles: PrivatePlayer | undefined;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishDisplayname:string[] = []; // catfish's dislplay name used in the html lobbyCode
+  let catfishList:string = "";
+
+  onMount (async () => {
+    lobbyData = await getLobbyData(lobbyCode);
+
+    let aliveCount = 0;
+    let aliveCatCount = 0;
+    // function to count the number of players currently alive
+    if (lobbyData !== undefined) {
+      for (let i = 0; i < lobbyData.uids.length; i++) {
+        if (lobbyData.players[i].alive == true) {
+          aliveCount++;
+          // count the number of remaining players who are alive
+          lobbyRoles = await getPrivatePlayer(lobbyCode, lobbyData.uids[i]);
+          if (lobbyRoles !== undefined) {
+            if (lobbyRoles.role == "CAT") {
+              aliveCatCount++;
+              // within the alive players, count the number that are cats
+            } else {
+              catfishDisplayname[i] = lobbyData.players[i].displayName; // create a list of catfishes
+            }
+          }
+        }
+      }
+    }
+    let aliveCatFishCount = aliveCount - aliveCatCount;
+    catfishList = [catfishDisplayname.slice(0, -1).join(', '), catfishDisplayname.slice(-1)[0]].join(catfishDisplayname.length < 2 ? '' : ' and ');
```

Comment:
you might want to consider `Intl.ListFormat()` which makes english lists for you https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat/ListFormat